### PR TITLE
Added ExtensionMethods to codegen for calling service methods in IEnumerable<Entity>

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Proposed change

The codegen now creates ExtensionMethods to call services on IEnumerable<SomeEntity> to enable calling services on multiple entities at a time

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/HassModel/NetDeamon.HassModel/Entities/EnumerableEntityExtensions.cs [hunk 1]
```diff
@@ -34,19 +34,35 @@ public static class EnumerableEntityExtensions
...
+        
+        entities = entities.ToList();
+        
+        if (!entities.Any()) return;
+        
+        var (serviceDomain, serviceName) = service.SplitAtDot();
+        
+        // Usually each Entity will have the same IHaContext and domain, but just in case its not, group by the context and domain and call the
+        // service for each group separately
+        var serviceCalls = entities.GroupBy(e => (
```
Comment (helto4real) [inline src/HassModel/NetDeamon.HassModel/Entities/EnumerableEntityExtensions.cs:56]: Not sure about this my self but will there be use-case where the order in the list are important? Ordering is hidden from users in this case.

File: src/HassModel/NetDeamon.HassModel/Entities/EntityExtensions.cs [hunk 2]
```diff
@@ -56,4 +56,21 @@ public static NumericEntity<TAttributes> WithAttributesAs<TAttributes>(this Nume
 
     internal static IObservable<T> StateChangesOnly<T>(this IObservable<T> changes) where T : StateChange
         => changes.Where(c => c.New?.State != c.Old?.State);
+    
+    public static IObservable<StateChange> SameStateFor(this IObservable<StateChange> observable, Func<EntityState?, bool> predicate, TimeSpan timeSpan) =>
```
Comment (helto4real) [inline src/HassModel/NetDeamon.HassModel/Entities/EntityExtensions.cs:60]: A xml comment on this too?

**Review by helto4real — APPROVED** [review]
Just some minor things and a question. Looks good!

File: src/HassModel/NetDeamon.HassModel/Entities/EntityExtensions.cs (hunk 2 above)
Comment (FrankBakkerNl) [reply src/HassModel/NetDeamon.HassModel/Entities/EntityExtensions.cs:60]: This was not supposed to be in this PR

**Pull request merged**